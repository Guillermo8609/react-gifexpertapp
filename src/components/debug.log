[0928/091409.040:ERROR:registration_protocol_win.cc(103)] CreateFile: El sistema no puede encontrar el archivo especificado. (0x2)
