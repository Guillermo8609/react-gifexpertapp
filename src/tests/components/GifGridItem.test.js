import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem/>', () => {
    const title =  'Un tiutlo'
    const url = 'https://localhost/algo.png'
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )
    
    test('Debe mostrar el componente correctamente ', () => {


        expect( wrapper ).toMatchSnapshot();


    })

    test('Debe tener un parrafo en el title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( title );

        
    })

    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        //console.log(img.prop('src'))
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
        
    })

    test('Debe tener animate__bounceIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className')

        expect( className.includes('animate__bounceIn')).toBe(true)
        
    })
    
    
        


})
