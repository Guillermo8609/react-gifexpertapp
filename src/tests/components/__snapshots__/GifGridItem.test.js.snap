// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Un tiutlo"
    src="https://localhost/algo.png"
  />
  <p>
    Un tiutlo
  </p>
</div>
`;
